catalog/tests/factories/github/__init__.py

```diff
@@ -152,3 +179,109 @@ def make_branch_protection(required_reviewers: int = 2) -> dict:
     ] = required_reviewers
 
     return branch_protection
+
+
+EventsConfig = namedtuple("EventsConfig", "event_name, days_since_previous_event")
+
+
+def make_events(origin_days_ago: int, events: list[str | EventsConfig]) -> list[dict]:
+    """
+    Create an issue events list with events starting at the origin.
+
+    The complexity of this fixture function is necessary to ease building
+    events list and testing the various states an events list can be in.
+
+    :param origin_days_ago: The number of days ago to treat as the origin
+    for the events list. If 0 then all events will be treated as being created the
+    same day and days lapsed configured for individual events will be ignored.
+    :param events: A list of mixed strings and (string, int) tuples. The string
+    must always be an event name. If a tuple, the int should be the number
+    of days to increment the event past the previous event.
+    """
+    result = []
+    previous_event_days_ago = origin_days_ago
+    for event_name in events:
+        days_since_previous_event = 0
+        if isinstance(event_name, tuple):
+            if origin_days_ago == 0:
+                # If origin days is 0 then treat all events as occurring today,
+                # otherwise events configured with days lapsed would end up
+                # in the future.
+                days_since_previous_event = 0
+            else:
+                days_since_previous_event = event_name[1]
+
+            event_name = event_name[0]
+
+        # Example: If the previous event was 2 days ago, and it has been
+        # 1 day since that previous event, then this event was 1 day ago.
+        days_ago = previous_event_days_ago - days_since_previous_event
+        created_at = walk_backwards_in_time_until_weekday_count(
+            datetime.datetime.now(), days_ago
+        )
+
+        event = _read_fixture(f"events/{event_name}")
+        event["created_at"] = _gh_date(created_at)
+
+        previous_event_days_ago = days_ago
+        result.append(event)
+
+    return result
+
+
+def make_non_urgent_events(events: list[str | EventsConfig]) -> list[dict]:
+    """
+    Create an events list where all events happened today.
+
+    By setting ``origin_days_ago`` to 0 we force the entire
+    events list to happen _now_ which necessarily means
+    none of the events could be considered urgent.
+    """
+    return make_events(0, events)
+
+
+def make_non_urgent_reviewable_events(events: list[str | EventsConfig]) -> list[dict]:
```

Comment:
Previously this was hard-coding `origin_days_ago` to 0, which forces all the events to happen "today". This does have the effect of ensuring the events list is non-urgent, but it collapsed the list unnecessarily. All we need to really do is ensure that the _last `ready_for_review` event _ happened today.

By updating this to do so, we can add an `events_list` to the parametrized list that has an initial `ready_for_review` event a very long time ago, so that _all_ of our tests will effectively test that only the date of the latest ready_for_review event is considered when checking PRs. (You can verify this: if you remove the [break](https://github.com/WordPress/openverse/pull/2027/files#diff-586c0dab45307447dd65e29600c995a9f4b85cad75f1c241b6f67f8e31d99e3bR98) and run the tests, many will fail.)